src/rgw/rgw_lua.cc

```diff
@@ -133,40 +133,49 @@ int add_package(const DoutPrefixProvider *dpp, rgw::sal::Driver* driver, optiona
     return ret;
   }
 
-  auto lua_mgr = driver->get_lua_manager();
-
-  return lua_mgr->add_package(dpp, y, package_name);
+  return driver->get_lua_manager("")->add_package(dpp, y, package_name);
 }
 
 int remove_package(const DoutPrefixProvider *dpp, rgw::sal::Driver* driver, optional_yield y, const std::string& package_name)
 {
-  auto lua_mgr = driver->get_lua_manager();
-
-  return lua_mgr->remove_package(dpp, y, package_name);
+  return driver->get_lua_manager("")->remove_package(dpp, y, package_name);
 }
 
 namespace bp = boost::process;
 
 int list_packages(const DoutPrefixProvider *dpp, rgw::sal::Driver* driver, optional_yield y, packages_t& packages)
 {
-  auto lua_mgr = driver->get_lua_manager();
+  return driver->get_lua_manager("")->list_packages(dpp, y, packages);
+}
 
-  return lua_mgr->list_packages(dpp, y, packages);
+namespace fs = std::filesystem;
+
+// similar to to the "mkdir -p" command
+int create_directory_p(const DoutPrefixProvider *dpp, const fs::path& p) {
+  std::error_code ec;
+  fs::path total_path;
+  for (const auto& pp : p) {
+    total_path /= pp;
+    auto should_create = !fs::exists(total_path, ec);
```

Comment:
can we not just call `create_directory()` and ignore `EEXIST` errors?

Reply:
`mkdir -p` does two things: ignore directories that already exists and create parent directories if missing